File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    var ans = [];
    for (c of commands) {
        var div_arr = array.slice(c[0]-1, c[1]);
        div_arr.sort((a, b) => a-b);
        ans.push(div_arr[c[2]-1]);
    }
    return ans;
}

File: 프로그래머스/1/42748. K번째수/K번째수.py
def quick_sort(arr):
    if len(arr) <= 1:
        return arr
    
    pivot = arr[len(arr) // 2]
    left, equal, right = [], [], []
    
    for num in arr:  # 오름차순 정렬
        if num < pivot:
            left.append(num)
        elif num > pivot:
            right.append(num)
        else:
            equal.append(num)
    
    return quick_sort(left) + equal + quick_sort(right)

def solution(array, commands):
    ans = []
    for i, j, k in commands:
        sorted_arr = quick_sort(array[i-1:j])
        ans.append(sorted_arr[k-1])
        
    return ans